### PR TITLE
adding lixtools recipe

### PR DESCRIPTION
**Pull request by kbeeperez**
implements: 
https://github.com/nsls-ii-forge/lixtools-feedstock
https://github.com/nsls-ii-forge/py4xs-feedstock

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kbeeperez** [comment]
Attn @mrakitin

File: recipes/lixtools/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    - py4xs
+    - python-barcode
+    - qrcode
+
+test:
+  imports:
+    - lixtools
+  commands:  # [not win]
+    - python -c 'import os, pkg_resources; assert os.path.isfile(pkg_resources.resource_filename("lixtools", "plate_label_template.html"))'  # [not win]
+    - python -c 'import os, pkg_resources; assert os.path.isfile(pkg_resources.resource_filename("lixtools", "template_report.ipynb"))'  # [not win]
```
Comment (mrakitin) [inline recipes/lixtools/meta.yaml:38]: I am against excluding tests for Windows. Could you try to fix it instead? On Windows CLI, single quotes are not accepted. Please use double-quotes. An easy way to test it on your Windows machine is to install the package locally and try this `python -c ...` string.

File: recipes/lixtools/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+about:
+  home: https://github.com/NSLS-II-LIX/lixtools
+  license: BSD-3-Clause
+  license_file: LICENSE
+  summary: Software tools for data collection/processing at NSLS-II LiX
+
+extra:
+  recipe-maintainers:
+    - mrakitin
```
Comment (mrakitin) [inline recipes/lixtools/meta.yaml:48]: Adding @lyang11973 as a core developer of the package. Lin, do you agree to be a maintainer of this conda-forge feedstock?
```suggestion
    - lyang11973
    - mrakitin
```

File: recipes/py4xs/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+about:
+  home: https://pypi.org/project/py4xs
+  license: MIT
+  license_file: LICENSE.txt
+  summary: py4xs - a python package for processing x-ray scattering data
+  dev_url: https://github.com/NSLS-II-LIX/py4xs
+
+extra:
+  recipe-maintainers:
+    - mrakitin
```
Comment (mrakitin) [inline recipes/py4xs/meta.yaml:44]:
```suggestion
    - lyang11973
    - mrakitin
```

**Review by mrakitin — CHANGES_REQUESTED** [review]
Looks good overall, I added a comment on how to solve the Windows tests that are currently excluded.

File: recipes/lixtools/meta.yaml [hunk 4]
```diff
@@ -33,9 +33,9 @@ requirements:
 test:
   imports:
     - lixtools
-  commands:  # [not win]
-    - python -c 'import os, pkg_resources; assert os.path.isfile(pkg_resources.resource_filename("lixtools", "plate_label_template.html"))'  # [not win]
-    - python -c 'import os, pkg_resources; assert os.path.isfile(pkg_resources.resource_filename("lixtools", "template_report.ipynb"))'  # [not win]
+  commands:
+    - python -c "import os, pkg_resources; assert os.path.isfile(pkg_resources.resource_filename("lixtools", "plate_label_template.html"))"
+    - python -c "import os, pkg_resources; assert os.path.isfile(pkg_resources.resource_filename("lixtools", "template_report.ipynb"))"
```
Comment (mrakitin) [inline recipes/lixtools/meta.yaml:38]: The double quotes inside should be replaced with single quotes or escaped to properly format the command.

```suggestion
    - python -c "import os, pkg_resources; assert os.path.isfile(pkg_resources.resource_filename('lixtools', 'plate_label_template.html'))"
    - python -c "import os, pkg_resources; assert os.path.isfile(pkg_resources.resource_filename('lixtools', 'template_report.ipynb'))"
```

**Review by mrakitin — APPROVED** [review]
Looks good to me!

@conda-forge/staged-recipes, I agree to be a maintainer of these feedstocks/packages.

File: recipes/lixtools/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    - openpyxl >=3
+    - pandas
+    - pillow
+    - py4xs
+    - python-barcode
+    - qrcode
+
+test:
+  imports:
+    - lixtools
```
Comment (dopplershift) [inline recipes/lixtools/meta.yaml:35]: Can you add some better import tests (`__init__` only sets the version)? I see a whole bunch of other imports in other modules that aren't listed above, like dask and scipy.

File: recipes/lixtools/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python >=3.6
+    - matplotlib
```
Comment (dopplershift) [inline recipes/lixtools/meta.yaml:24]:
```suggestion
    - matplotlib-base
```

File: recipes/py4xs/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    - fabio
+    - h5py
+    - matplotlib
+    - numpy
+    - pillow
+    - scipy
+
+test:
+  imports:
+    - py4xs
```
Comment (dopplershift) [inline recipes/py4xs/meta.yaml:33]: Same here, `__init__` does nothing so importing some other modules would be a much better test that things are right here.

File: recipes/py4xs/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python >=3.6
+    - fabio
+    - h5py
+    - matplotlib
```
Comment (dopplershift) [inline recipes/py4xs/meta.yaml:26]:
```suggestion
    - matplotlib-base
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
@lyang11973 are you ok with being listed as a maintainer for both of these recipes?

**Comment by lyang11973** [comment]
Not sure what I need to do. But sure, I agree to be a maintainer.

**Comment by dopplershift** [comment]
@lyang11973 You'll receive GitHub notifications about issues and PRs on the repository that holds the recipe (the information that's used to build a package for lixtools on conda-forge).

**Review by dopplershift — APPROVED** [review]

**Comment by mrakitin** [comment]
@dopplershift, do you know why the final checks before merging did not run the tests? The resulting feedstock with more tests (added here via https://github.com/conda-forge/staged-recipes/pull/15648/commits/5b2c036357baba49c73af4b2ab4ad29a96158d84) uncovered missing dependencies (in particular, `opentrons`, that I am working on within https://github.com/conda-forge/staged-recipes/pull/15705).

**Comment by dopplershift** [comment]
No, and 🐑 that I missed it.

**Pull request merged**